File: 5 PrefixSums/PassingCars/passingCars.js
function solution(A) {
  const max = 1000000000;
  let lane = 0;
  let sum = 0;
  for (let a of A) {
    if (a === 0) lane += 1;
    else sum += lane;
    if (sum > max) return -1;
  }
  return sum;
}

console.log(solution([0, 1, 0, 1, 1]));
console.log(solution([0]));
//O(N)

File: Lessons/5 PrefixSums/PassingCars/passingCars.cpp
int solution(vector<int> &A) {
  int max = 1000000000;
  int lane = 0;
  int sum = 0;
  for (int a : A) {
    if (a == 0)
      lane += 1;
    else
      sum += lane;
    if (sum > max) return -1;
  }
  return sum;
}
